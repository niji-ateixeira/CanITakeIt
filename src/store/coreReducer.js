// Ce fichier sert a regrouper les differant reducer a combin√©
// pour les ajouter dans le store par la suite
import { combineReducers } from 'redux';

import listDrugs from '../reducers/listDrugs';
// Liste des reducer a combiner pour les ajouter au store

// Combinaison des different store
export default combineReducers({ listDrugs });
