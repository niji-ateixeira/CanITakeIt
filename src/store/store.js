/*
  Fichier permettant la cr√©ation du store

  Date de modification: 28/08/2017
*/

import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';

import CoreReducer from './coreReducer';
import getDrugs from '../sagas/getDrugs';

const sagaMiddleware = createSagaMiddleware();
// Creation du store
const store = createStore(CoreReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(getDrugs);

export default store;
