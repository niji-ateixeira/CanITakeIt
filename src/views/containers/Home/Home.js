import React from 'react';
import { View, Image } from 'react-native';
import { Actions } from 'react-native-router-flux';

// action
import { Button, Text } from 'react-native-elements';
import logo from '../../../img/logo.png';
// Style
import styles from './HomeStyles';

const Home = () => (
  <View style={styles.container}>
    <Text h4 style={styles.title}>
          Bienvenue sur Notice
    </Text>
    <Image
      source={logo}
      style={{ width: 300, height: 75, marginBottom: 50 }}
    />
    <Button
      onPress={() => Actions.research()}
      backgroundColor="#00b3b3"
      icon={{ name: 'camera' }}
      title="Scanner votre mÃ©dicament"
    />
  </View>
);

export default Home;
