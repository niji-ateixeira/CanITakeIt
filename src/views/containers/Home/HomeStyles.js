import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#eefbf4',
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: { marginBottom: 30 },
});

export default styles;
