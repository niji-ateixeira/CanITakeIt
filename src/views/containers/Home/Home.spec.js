import React from 'react';

import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import Home from './Home';
import store from '../../../store/store';

test('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}>
    <Home />
                               </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
