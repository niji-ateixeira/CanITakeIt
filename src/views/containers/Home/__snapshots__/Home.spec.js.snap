// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eefbf4",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        Object {
          "fontSize": 27.5,
        },
        undefined,
        undefined,
        undefined,
        Object {},
        undefined,
        Object {
          "marginBottom": 30,
        },
      ]
    }
  >
    Bienvenue sur Notice
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/img/logo.png",
      }
    }
    style={
      Object {
        "height": 75,
        "marginBottom": 50,
        "width": 300,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#00b3b3",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Scanner votre médicament
        </Text>
      </View>
    </View>
  </View>
</View>
`;
